app/components/Nav/App/index.js

```diff
@@ -762,6 +766,46 @@ const App = (props) => {
   useEffect(() => {
     async function startApp() {
       const existingUser = await StorageWrapper.getItem(EXISTING_USER);
+      if (!existingUser) {
+        // List of chainIds to add (as hex strings)
+        const chainIdsToAdd = [
+          CHAIN_IDS.ARBITRUM,
+          CHAIN_IDS.BASE,
+          CHAIN_IDS.BSC,
+          CHAIN_IDS.OPTIMISM,
+          CHAIN_IDS.POLYGON,
+        ];
+
+        // Filter the PopularList to get only the specified networks based on chainId
+        const selectedNetworks = PopularList.filter((network) =>
+          chainIdsToAdd.includes(network.chainId),
+        );
+        const { NetworkController } = Engine.context;
+
+        // Loop through each selected network and call NetworkController.addNetwork
+        for (const network of selectedNetworks) {
+          try {
+            await NetworkController.addNetwork({
+              chainId: network.chainId,
+              blockExplorerUrls: [network.rpcPrefs.blockExplorerUrl],
+              defaultRpcEndpointIndex: 0,
+              defaultBlockExplorerUrlIndex: 0,
+              name: network.nickname,
+              nativeCurrency: network.ticker,
+              rpcEndpoints: [
+                {
+                  url: network.rpcUrl,
+                  name: network.nickname,
+                  type: RpcEndpointType.Custom,
+                },
+              ],
+            });
+          } catch (error) {
+            Logger.error(error);
+          }
+        }
+      }
```

Comment:
I wonder if `networksToAdd` is more accurate, because the networks are not really selected, wdyt?

```suggestion
        const networksToAdd = PopularList.filter((network) =>
          chainIdsToAdd.includes(network.chainId),
        );
        const { NetworkController } = Engine.context;

        // Loop through each selected network and call NetworkController.addNetwork
        for (const network of networksToAdd) {
          try {
            await NetworkController.addNetwork({
              chainId: network.chainId,
              blockExplorerUrls: [network.rpcPrefs.blockExplorerUrl],
              defaultRpcEndpointIndex: 0,
              defaultBlockExplorerUrlIndex: 0,
              name: network.nickname,
              nativeCurrency: network.ticker,
              rpcEndpoints: [
                {
                  url: network.rpcUrl,
                  name: network.nickname,
                  type: RpcEndpointType.Custom,
                },
              ],
            });
          } catch (error) {
            Logger.error(error);
          }
        }
      }
```